go-controller/pkg/ovn/obj_retry.go

```diff
@@ -436,14 +488,29 @@ func (oc *Controller) recordDeleteEvent(objType reflect.Type, obj interface{}) {
 	}
 }
 
-// Given an object and its type, recordErrorEvent records an error event on this object. Only used for pods now.
-func (oc *Controller) recordErrorEvent(objType reflect.Type, obj interface{}, err error) {
+// Given an object and its type, recordErrorEvent records an error event on this object.
+func (oc *Controller) recordErrorEvent(objType reflect.Type, obj, newerObj interface{}, err error) {
 	switch objType {
 	case factory.PodType:
 		klog.V(5).Infof("Recording error event on pod")
 		pod := obj.(*kapi.Pod)
 		oc.metricsRecorder.AddPod(pod.UID)
 		oc.recordPodEvent(err, pod)
+	case factory.EgressFirewallType:
+		egressFirewall := obj.(*egressfirewall.EgressFirewall).DeepCopy()
+		if newerObj == nil {
+			egressFirewall.Status.Status = egressFirewallAddError
+			metrics.UpdateEgressFirewallRuleCount(float64(len(egressFirewall.Spec.Egress)))
+			metrics.IncrementEgressFirewallCount()
```

Comment:
why are you messing with metrics in recordErrorEvent? Seems like we shouldn't be messing with that here. Cant we just increment/decrement metrics in addEgressFirewall or deleteEgressFirewall?

Reply:
u are right add and delete are sufficient to handle metrics I complicated it for no reason will clean this up